app/controllers/flows_controller.rb

```diff
@@ -29,14 +29,25 @@ def show
 
   private
 
-  def screenshot_base
-    if Rails.env.development?
+  def screenshot_path(controller)
+    [screenshot_base, I18n.locale, "#{controller.name}.png"].join('/')
```

Comment:
I think it's better to check on disk then if not, provide the S3 URL, and only do that in dev.